pkg/storage/image.go

```diff
@@ -598,13 +606,21 @@ func GetImageService(ctx context.Context, sc *types.SystemContext, store storage
 		imageCache:            make(map[string]imageCacheItem),
 		ctx:                   ctx,
 	}
+
 	if len(registries) != 0 {
 		seenRegistries := make(map[string]bool, len(registries))
 		cleanRegistries := []string{}
 		for _, r := range registries {
 			if seenRegistries[r] {
 				continue
 			}
+			registry, err := sysregistriesv2.FindRegistry(nil, r)
```

Comment:
I can’t concentrate to review this in detail now, but shouldn’t the “search registries” case also use the full image name? (i.e. it would have to happen in the search evaluation phase, not here)

Reply:
that's already covered in `ResolveNames` then right?

Reply:
I'm removing this bits from here

Reply:
`ResolveNames` does not currently call `FindRegistry` on the full path, i.e. the value `path.Join(r, rem)`